src/js/helpers/graph/declarationHelper.js

```diff
@@ -44,24 +56,25 @@ const createVisNode = (node) => ({
   group: getNodeGroup(node),
 });
 
-const getLeafs = (root, depth = 0) => {
+const getEndNodes = (root, depth = 0) => {
   if (!hasTransitions(root)) {
     return { node: root, depth };
   }
 
-  return Object.entries(root.on).flatMap(([, child]) => getLeafs(child, depth + 1));
+  if (!hasAllCompositeEndTransitions(root) && root.type !== 'virtual_start') {
+    return [
+      ...Object.entries(root.on).flatMap(([, child]) => getEndNodes(child, depth + 1)),
+      { node: root, depth },
+    ];
+  }
+
+  return Object.entries(root.on).flatMap(([, child]) => getEndNodes(child, depth + 1));
```

Comment:
You could have used Object.values instead of Object.entries
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values